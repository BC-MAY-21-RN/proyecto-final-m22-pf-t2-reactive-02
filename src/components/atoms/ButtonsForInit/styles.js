import {StyleSheet, Dimensions} from 'react-native';

const styles = StyleSheet.create({
  button1: {
    width: Dimensions.get('screen').width - 84,
    height: 45,
    backgroundColor: '#DB6C9E',
    borderRadius: 8,
    alignItems: 'center',
    justifyContent: 'center',
    marginLeft: 42,
    marginTop: 15,
  },
  button2: {
    width:
      Dimensions.get('screen').width <= 360
        ? Dimensions.get('screen').width - 120
        : Dimensions.get('screen').width - 158,
    height: 50,
    backgroundColor: '#fff',
    borderWidth: 3,
    borderColor: '#DB6C9E',
    borderRadius: 8,
    marginLeft: Dimensions.get('screen').width <= 360 ? 60 : 79,
    marginTop: 20,
    justifyContent: 'center',
  },
  text1: {
    color: '#fff',
    fontSize: 16,
  },
  text2: {
    color: '#DB6C9E',
    fontSize: 20,
    fontWeight: '600',
    marginLeft: 35,
  },
  google: {
    position: 'absolute',
    right: 35,
  },
});

export default styles;
