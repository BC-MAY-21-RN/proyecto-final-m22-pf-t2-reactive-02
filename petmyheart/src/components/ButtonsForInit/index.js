import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import styles from './styles';
import GoogleSVG from '../../assets/icons/google.svg';
import Icons from '../Icons';

export default function ButtonsForInit({}) {
  return (
    <View>
      <TouchableOpacity style={styles.button1}>
        <Text style={styles.text1}>CREAR CUENTA</Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.button2}>
        <Text style={styles.text2}>Ingresa con:</Text>
        <Icons IconProp={GoogleSVG} style={styles.google} />
      </TouchableOpacity>
    </View>
  );
}
