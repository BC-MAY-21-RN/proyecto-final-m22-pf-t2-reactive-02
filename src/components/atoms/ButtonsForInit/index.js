import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import styles from './styles';
import GoogleSVG from '../../../assets/icons/google.svg';
import Icons from '../Icons';
import auth from '@react-native-firebase/auth';
import firestore from '@react-native-firebase/firestore';
import {GoogleSignin} from '@react-native-google-signin/google-signin';

const addUser = (user, changeLoading) => {
  firestore()
    .collection('usuarios')
    .add({
      correo: user.valuesRegister.email,
      nombre: user.valuesRegister.name,
      uid: auth().currentUser.uid,
    })
    .then(() => {
      changeLoading(false);
    });
};

const registerEmailUser = (user, changeLoading) => {
  auth()
    .createUserWithEmailAndPassword(
      user.valuesRegister.email,
      user.valuesRegister.password,
    )
    .then(() => {
      addUser(user, changeLoading);
    })
    .catch(error => {
      changeLoading(false);
    });
};

const validateData = (user, changeAlerts, changeLoading) => {
  const list = [false, false, false, false];
  const inputs = [
    {fun: user.validateName(user)},
    {fun: user.validateEmail(user)},
    {fun: user.validatePassword(user)},
    {fun: user.validatePassword2(user)},
  ];

  inputs.forEach((item, index) => {
    list[index] = item.fun;
  });
  changeAlerts(list);

  if (list.includes(true)) {
    changeLoading(false);
  } else {
    changeLoading(true);
    registerEmailUser(user, changeLoading);
  }
};

async function onGoogleButtonPress(changeLoading) {
  changeLoading(true);
  const {idToken} = await GoogleSignin.signIn();
  const googleCredential = auth.GoogleAuthProvider.credential(idToken);
  return auth().signInWithCredential(googleCredential);
}

const validateDataGoogle = changeLoading => {
  const uid = auth().currentUser.uid;
  const user = {
    valuesRegister: {
      name: auth().currentUser.displayName,
      email: auth().currentUser.email,
    },
  };
  const db = firestore();
  db.collection('usuarios')
    .where('uid', '==', uid)
    .get()
    .then(querySnapshot => {
      if (querySnapshot.size === 0) {
        addUser(user, changeLoading);
      }
    });
};

const loginAccount = (email, password, changeLoading) => {
  changeLoading(true);
  auth()
    .signInWithEmailAndPassword(email, password)
    .then(() => {
      changeLoading(false);
    })
    .catch(error => {
      changeLoading(false);
    });
};

export default function ButtonsForInit({user, changeAlerts, changeLoading}) {
  return (
    <View>
      <TouchableOpacity
        disabled={!user?.getBool()}
        style={{
          ...styles.button1,
          ...(user?.getBool() ? {} : {backgroundColor: '#B09AAC'}),
        }}
        onPress={() =>
          changeAlerts === undefined
            ? loginAccount(
                user.valuesLogin.email,
                user.valuesLogin.password,
                changeLoading,
              )
            : validateData(user, changeAlerts, changeLoading)
        }>
        <Text style={styles.text1}>
          {' '}
          {changeAlerts === undefined ? 'INICIAR SESIÃ“N' : 'CREAR CUENTA'}
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        disabled={!user?.getBoolGoogle()}
        style={{
          ...styles.button2,
          ...(user?.getBoolGoogle() ? {} : {borderColor: '#B09AAC'}),
        }}
        onPress={() => {
          onGoogleButtonPress(changeLoading)
            .then(() => validateDataGoogle(changeLoading))
            .catch(() => changeLoading(false));
        }}>
        <Text
          style={{
            ...styles.text2,
            ...(user?.getBoolGoogle() ? {} : {color: '#B09AAC'}),
          }}>
          Ingresa con:
        </Text>
        <Icons IconProp={GoogleSVG} style={styles.google} />
      </TouchableOpacity>
    </View>
  );
}
