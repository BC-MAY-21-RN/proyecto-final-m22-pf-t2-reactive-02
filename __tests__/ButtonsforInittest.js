import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ButtonsForInit from '../src/components/atoms/ButtonsForInit';

describe('Test from ButtonsForInit component', () => {
  it('ButtonsForInit snapshot test', () => {
    const renderButtonsForInit = renderer
      .create(
        <ButtonsForInit
          onPress1={() => {}}
          onPress2={() => {}}
          textButton={''}
          validationbtn1={true}
          validationbtn2={true}
          key={1}
        />,
      )
      .toJSON();
    expect(renderButtonsForInit).toMatchSnapshot();
  });
});
