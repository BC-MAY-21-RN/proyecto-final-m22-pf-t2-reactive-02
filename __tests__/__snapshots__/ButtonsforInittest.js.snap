// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test from ButtonsForInit component ButtonsForInit snapshot test 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#DB6C9E",
        "borderRadius": 8,
        "height": 45,
        "justifyContent": "center",
        "marginLeft": 42,
        "marginTop": 15,
        "opacity": 1,
        "width": 666,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 16,
        }
      }
    />
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#DB6C9E",
        "borderRadius": 8,
        "borderWidth": 3,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "marginLeft": 79,
        "marginTop": 20,
        "opacity": 1,
        "width": 592,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#DB6C9E",
          "fontSize": 20,
          "fontWeight": "600",
        }
      }
    >
      INGRESA CON: 
    </Text>
    <View>
      < />
    </View>
  </View>
</View>
`;
